src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -686,12 +686,6 @@ public static bool IsAutoPropertyAccessor(this IMethodSymbol methodSymbol)
         /// </summary>
         public static bool IsTopLevelStatementsEntryPointMethod([NotNullWhen(true)] this IMethodSymbol? methodSymbol)
             => methodSymbol?.ContainingType.IsTopLevelStatementsEntryPointType() == true &&
-               methodSymbol.IsStatic &&
-               methodSymbol.Name switch
-               {
-                   "$Main" => true,
-                   "<$Main>$" => true,
-                   _ => false
-               };
+               methodSymbol.IsStatic && methodSymbol.Name == "<Main>$";
```

Comment:
Please retain the old logic as `"$Main"` was used in older compiler bits which is supported by the analyzer package. I think you should only fix the typo here.

Reply:
I thought that's not important as the compiler version has this isn't a stable release. I'll revert the last 2 commits.

Reply:
> as the compiler version has this isn't a stable release

Ah, if you are sure it wasn't a stable release, then it would be fine to remove those cases. By the way, there are test failures.

Reply:
> Ah, if you are sure it wasn't a stable release, then it would be fine to remove those cases

@mavasani I think it was shipped only in a preview version of Visual Studio. But might be better to confirm with the compiler team. Not sure who may be able help.

Reply:
@mavasani The current test failures seems unrelated to top-level statements. Can you have a look?

For example:

https://github.com/dotnet/roslyn-analyzers/blob/a23fde8d03fcec0c8e5beb6b1f3882048ab8435b/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ImplementIDisposableCorrectlyTests.cs#L745-L798

This test doesn't produce any diagnostics, while 2 are expected. The number of failing tests is **huge**.